client/forest-manager/src/prove.rs

```diff
@@ -46,122 +46,57 @@ pub(crate) fn prove<T: TrieLayout, F: ForestStorage>(
         .map_err(|_| ForestStorageErrors::FailedToReadLeaf)?;
 
     match (prev, next) {
-        (_, Some((key, value))) if challenged_file_key.as_ref() == key => {
-            // Scenario 1: Exact match
-            Ok(Proven::new_exact_key(
-                key.into(),
-                deserialize_value(&value)?,
-            ))
-        }
-        (Some((prev_key, prev_value)), Some((next_key, next_value))) => {
-            // Scenario 2: Between two keys
-            let prev_leaf = Leaf {
-                key: prev_key.into(),
-                data: deserialize_value(&prev_value)?,
-            };
-            let next_leaf = Leaf {
-                key: next_key.into(),
-                data: deserialize_value(&next_value)?,
-            };
+        // Scenario 1: Exact match
+        (_, Some((key, value))) if challenged_file_key.as_ref() == key => Ok(
+            Proven::new_exact_key(key.into(), deserialize_value(&value)?),
+        ),
+        // Scenario 2: Between two keys
+        (Some((prev_key, prev_value)), Some((next_key, next_value)))
+            if prev_key < challenged_file_key.as_ref().to_vec()
+                && next_key > challenged_file_key.as_ref().to_vec() =>
+        {
+            let prev_leaf = Leaf::new(prev_key.into(), deserialize_value(&prev_value)?);
+            let next_leaf = Leaf::new(next_key.into(), deserialize_value(&next_value)?);
+
             Ok(Proven::new_neighbour_keys(Some(prev_leaf), Some(next_leaf))
                 .map_err(|_| ForestStorageErrors::FailedToConstructProvenLeaves)?)
         }
-        (Some((key, value)), None) if *challenged_file_key.as_ref() > *key => {
-            // Scenario 3: After the last leaf
-            let leaf = Leaf {
-                key: key.into(),
-                data: deserialize_value(&value)?,
-            };
-            Ok(Proven::new_neighbour_keys(Some(leaf), None)
+        // Scenario 3: Before the first leaf
```

Comment:
I wouldn't want to merge this PR with this work around to an inconsistent behaviour of trie-db. Let's leave only the match clause of what would be the expected behaviour, and leave ignored tests to make sure this behaviour gets fixed in the future.

Reply:
Agreed.